### ISSUE TITLE
Plugin asks for bluetooth permission on app startup (iOS)

### ISSUE THREAD

**Issue by adamk22** [pauldemarco/flutter_blue#696]
I'm using this plugin in an app where bluetooth is optional. However, in iOS, when the user boots up the app for the first time the plugin immediately asks for bluetooth permissions. Is it possible to control when to ask a permission? For instance, in a view where the bluetooth feature is utilized? 

I came across issue #272 with no clear solution other than digging into the source code and I was wondering if anyone has a solution?

**Comment by MrCsabaToth** [comment]
Duplicate of #696

**Comment by ollydixon** [comment]
Yeah this is a serious problem for us too. We'd like to be able to ask permission and explain why we need it too users before it actually asks them. Currentely it just pops up right away. This should be an easy fix, not sure how to implement it though (iOS)

**Comment by anslex** [comment]
I agree that it is a serious issue to publish an app to market
Please improve this functionnality

**Comment by MrCsabaToth** [comment]
Let's just first fix the even more serious issue of  #726

**Comment by Chris-FizziQ** [comment]
Hi, I have this issue as well and it is quite unfortunate as it spoils the ergonomic of an app. Have any of you found some kind of solution ? Many thanks.

**Comment by vaibhiarora03** [comment]
@Chris-FizziQ  check #599 .

**Comment by Manasmd** [comment]
Do we have a fix from the package itself without having to change the IOS code for this?
